### PR TITLE
Support ignored tags as preprocessing pass

### LINKED ISSUE #261: Ignored tags around attributes gives an error
Files like:
&lt;% if (true) { %&gt;
&lt;div class="test"&gt;&lt;/div&gt;
&lt;%}%&gt;

works fine, and the contents of the server tags are ignored, but if I have the server tags around an attribute , they are not ignored:

&lt;div class="&lt;%=test||"en"%>"&gt;&lt;/div&gt;

this fails because of quotes inside quotes, but if would make more sense if the code was just ignored, just like when they are outside a tag.

### PR DESCRIPTION
**Pull request by duncanbeevers**
Fixes #261

### CODE REVIEW

**Comment by duncanbeevers** [comment]
This patch also provided ~5% speedup on the wikipedia benchmark.

**Comment by duncanbeevers** [comment]
The change to the existing test is due to ignored tags no longer being handled as tags per-se, but instead as opaque strings. After they're removed, the ordinary whitespace collapsing rules come into play, and those rules preclude collapsing out the central `whitespace` in this construct: `text``whitespace``text`

**Comment by duncanbeevers** [comment]
This could be pretty easily adapted to deal with #83 as well.

File: src/htmlminifier.js [hunk 1]
```diff
@@ -534,6 +534,12 @@
         isIgnoring = false,
         t = new Date();
 
+    if (options.removeIgnored) {
+      value = value.
```
Comment (kangax) [inline src/htmlminifier.js]: can you please move `.` to the next line, for consistency

**Comment by duncanbeevers** [comment]
Oh yeah, this also makes ignored-tag-removal a little more strict, demanding matching opening and closing tag pairs.

`<%` can only be closed by `%>` and `<?` can only be closed by `?>`.

**Comment by kangax** [comment]
@duncanbeevers thanks!

**Pull request merged**